app/parse.py

```diff
@@ -8,8 +16,51 @@ class Quote:
     tags: list[str]
 
 
+def get_single_quote(quote_soup: BeautifulSoup) -> Quote:
+    return Quote(
+        text=quote_soup.select_one(".text").text,
+        author=quote_soup.select_one(".author").text,
+        tags=[tag.text for tag in quote_soup.select(".tag")]
+    )
+
+
+def get_number_pages(quotes: Quote) -> int:
+    count_pages = 0
+    for _ in quotes:
+        count_pages += 1
+    return count_pages
```

Comment:
no need to make this like that, u can use len
also it is better just to check "next" button inside the while loop in ur get_quotes_per_page func